Fixes #16 blob value comparison

bffrost opened the pull request:
BlobValue comparison in `is_equal_ls_value_simple_value` was backwards in that it was taking attributes from a BlobValue and re-casting them again as a BlobValue, rather than taking attributes from an LsValue and forming a BlobValue.

Added a test of updating a blobValue to confirm the fix.

brianbolt commented:
Small nit is that the helper functions are built into the test, I tend to think if you are going to make a function you should make it available outside of the test or just make the code inline as it just makes the code a little harder to walk through.  Not a big deal, LGTM.

brianbolt commented:
Actually, ignore what I just said, I see you are reusing it a few times in the test so its definitely worth being a function. Just wonder if its useful outside this test but not reason to move it outside if it's not currently being used.

bffrost commented:
@brianbolt Thanks for the feedback. Yeah I made it a function to reuse a few times in the test. But you are right that it could be useful outside the test, so I've moved it and refactored the usages in the other test too.
